### PR TITLE
examples: modal with form

### PR DESCRIPTION
**Pull request by frassinier**
Add an example using a form in a modal.

image

It will increase the list of examples related to #626

### CODE REVIEW

**Comment by frassinier** [comment]
@diegohaz Does it fit? 🤓

File: packages/reakit/src/Dialog/__examples__/DialogWithForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  unstable_FormLabel as FormLabel,
+  unstable_FormInput as FormInput,
+  unstable_FormMessage as FormMessage,
+  unstable_FormSubmitButton as FormSubmitButton,
+} from "reakit/Form";
+
+export interface Props {
+  initialName: string;
+  onRename: (newName: string) => void;
+}
```
Comment (diegohaz) [inline packages/reakit/src/Dialog/__examples__/DialogWithForm/index.tsx]:
```suggestion
type Props = {
  initialName: string;
  onRename: (newName: string) => void;
};
```

File: packages/reakit/src/Dialog/__examples__/DialogWithForm/__tests__/index-test.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  expect(name).toBeVisible();
+  const disclosure = getByText("✏️");
+  const dialog = getByLabelText("Choose a new name");
+  expect(dialog).not.toBeVisible();
+  click(disclosure);
+  expect(dialog).toBeVisible();
+  const form = getByRole("form");
+  const input = getByLabelText("New name");
+  type(" 2", input);
+  fireEvent.submit(form);
```
Comment (diegohaz) [inline packages/reakit/src/Dialog/__examples__/DialogWithForm/__tests__/index-test.tsx]: I would try this (need to import `focus` and `press` from `reakit-test-utils`):
```suggestion
  focus(input);
  type(" 2");
  press.Enter();
```

File: packages/reakit/src/Dialog/__examples__/DialogWithForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    <Form {...form}>
+      <FormLabel {...form} name="name">
+        New name
+      </FormLabel>
+      <FormInput {...form} name="name" />
+      <FormMessage {...form} name="name" />
+      <FormSubmitButton {...form}>Rename</FormSubmitButton>
+    </Form>
+  );
+}
```
Comment (diegohaz) [inline packages/reakit/src/Dialog/__examples__/DialogWithForm/index.tsx:40]: Nice 👍

**Comment by frassinier** [comment]
Sure! I will merge all your suggestions, rebase master and add 2 or 3 examples, in addition. Thanks for your feedback!

**Comment by diegohaz** [comment]
Nice! You can add one example per PR. It'll be easier to review.

**Pull request merged**